app/src/main/java/com/wafflestudio/snutt2/data/lecture_search/LectureSearchRepositoryImpl.kt

```diff
@@ -66,6 +70,22 @@ class LectureSearchRepositoryImpl @Inject constructor(
         return response.content
     }
 
+    override fun storeRecentSearchedDepartment(tag: TagDto) {
+        storage.recentSearchedDepartments.update(
+            storage.recentSearchedDepartments.get().toMutableList().apply {
+                if (contains(tag)) remove(tag)
+                add(tag)
+                while (count() > 5) removeAt(0)
+            },
+        )
+    }
```

Comment:
```suggestion
    override fun storeRecentSearchedDepartment(tag: TagDto) {
        storage.recentSearchedDepartments.update(
            (storage.recentSearchedDepartments.get() + tag).takeLast(5),
        )
    }
```

Reply:
이거 코멘트 해준대로 고치면 이미 최근 5개에 있던 태그 검색했을 때
최근 검색에서 사라지기만 하는 문제점이 있어서, 아래처럼 고쳐봤는데 어때
```suggestion
    override fun storeRecentSearchedDepartment(tag: TagDto) {
        val previousStoredTags = storage.recentSearchedDepartments.get()

        storage.recentSearchedDepartments.update(
            (previousStoredTags.filter { it != tag } + tag).takeLast(5)
        )
    }
```

Reply:
혹시나 해서 public operator fun <T> Iterable<T>.minus(element: T): List<T> 읽어봤는데
리스트에 없던거 - 연산 해도 문제되는건 없네

Reply:
베리굿